In [2]:
import numpy as np

In [13]:
n_steps = "5k"
n_seed = 30
env_random = 0.1
q_lr = 0.1
r_lr = 0.1

eps = 1.0

lenght, rewards = np.zeros(n_seed), np.zeros(n_seed)
for seed in range(n_seed):
    log_dir = "../models_{}/9_9/Penalty/reward_model/env_{}/eps_{}/q_lr_{}/reward_lr_{}".format(n_steps, env_random, eps, q_lr, r_lr)
    traj = np.load(log_dir + "/trajectories_{}.npy".format(seed), allow_pickle=True)[()]
    gamma = 0.99
    ep_length = []
    ep_reward = []
    for key in traj.keys():
        ep_length.append(len(traj[key]["states"][:, 0]))
        ep_reward.append(np.sum(traj[key]["undiscounted_joint_reward"] * np.array([gamma**i for i in range(len(traj[key]["undiscounted_joint_reward"]))])))
    # print("episode length ", np.mean(ep_length), r"$\pm$", np.std(ep_length))
    # print("episode joint reward {:.3f} +- {:.3f}".format(np.mean(ep_reward), np.std(ep_reward)))
    lenght[seed], rewards[seed] = np.mean(ep_length), np.mean(ep_reward)
print("mean lenght {:.1f}, std = {:.2f}".format(np.mean(lenght), np.std(lenght)))
print("mean reward {:.3f}, std = {:.3f}".format(np.mean(rewards), np.std(rewards)))

mean lenght 26.6, std = 0.24
mean reward 0.616, std = 0.176


## Test rewad Mean & std for the greedy policy after 25M training steps across 30 seeds

### Env = 0.0, $Q_\alpha = 0.1$, $R_\alpha = 0.1$
. | $\epsilon = 0.1$ | $\epsilon = 0.5$ | $\epsilon = 0.7$ |$\epsilon = 1.0$ | $\epsilon$ = dec|
| --- | --- | --- |--- | --- | --- |
| 500 steps/episode | $0.0\pm0.0$ | $0.688\pm0.27$ | **$0.794\pm0.0$** | **$0.794\pm0.0$**| **$0.794\pm0.0$**|
| 5,000 steps/episode | $0.0\pm0.0$ | $0.556\pm0.364$ | **$0.794\pm0.0$** | **$0.794\pm0.0$**|**$0.794\pm0.0$** |


### Env = 0.1, $Q_\alpha = 0.1$, $R_\alpha = 0.1$
. | $\epsilon = 0.1$ | $\epsilon = 0.5$ | $\epsilon = 0.7$ |$\epsilon = 1.0$ | $\epsilon$ = dec|
| --- | --- | --- |--- | --- | --- |
| 500 steps/episode | $0.11\pm0.115$ | $-0.01\pm0.248$ | $-0.009\pm0.218$| $0.017\pm0.265$ | $0.3\pm0.265$|
| 5,000 steps/episode | $0.0\pm0.0$ |  $0.025\pm0.183$| $-0.002\pm0.215$ | **$0.616\pm0.176$**| $0.452\pm0.241$|

In [5]:
"../mo"

In [8]:
!ls

9_9_function_approximation.ipynb models
9_9_plots.ipynb                  plot_testing_reward.ipynb
evaluate_policy.ipynb


In [7]:
! cd ..